adal/src/main/java/com/microsoft/aad/adal/Logger.java

```diff
@@ -152,204 +192,225 @@ void Log(String tag, String message, String additionalMessage, LogLevel level,
+                //The debug level is deprecated and removed in common core.
+                com.microsoft.identity.common.internal.logging.Logger.info(tag, Logger.getInstance().mCorrelationId, formatMessage(message));
                 break;
             default:
-                throw new IllegalArgumentException("Unknown loglevel");
-        }
-    }
-
-    private static String addMoreInfo(String message) {
-        if (!StringExtensions.isNullOrBlank(message)) {
-            return getUTCDateTimeAsString() + "-" + getInstance().mCorrelationId + "-" + message
-                    + " ver:" + AuthenticationContext.getVersionName();
-        }
-
-        return getUTCDateTimeAsString() + "-" + getInstance().mCorrelationId + "- ver:"
-                + AuthenticationContext.getVersionName();
-    }
-
-    /**
-     * Format the log message. Depends on the developer setting, the log message could be sent to logcat
-     * or the external logger set by the calling app.
-     */
-    private void log(String tag, String message, String additionalMessage, LogLevel logLevel,
-                     ADALError errorCode, Throwable throwable) {
-        if (logLevel.compareTo(mLogLevel) > 0) {
-            return;
-        }
-
-        final StringBuilder logMessage = new StringBuilder();
-
-        if (errorCode != null) {
-            logMessage.append(getCodeName(errorCode)).append(':');
-        }
-
-        logMessage.append(addMoreInfo(message));
-
-        // Developer turns off PII logging, if the log message contains any PII, we shouldn't send it.
-        if (!StringExtensions.isNullOrBlank(additionalMessage) && mEnablePII) {
-            logMessage.append(' ').append(additionalMessage);
-        }
-
-        // Adding stacktrace to message
-        if (throwable != null) {
-            logMessage.append('\n').append(Log.getStackTraceString(throwable));
-        }
-
-        if (mAndroidLogEnabled) {
-            sendLogcatLogs(tag, logLevel, logMessage.toString());
-        }
-
-        if (mExternalLogger != null) {
-            try {
-                if (!StringExtensions.isNullOrBlank(additionalMessage) && mEnablePII) {
-                    mExternalLogger.Log(tag, addMoreInfo(message), additionalMessage + (throwable == null ? "" : Log.getStackTraceString(throwable)), logLevel, errorCode);
-                } else {
-                    mExternalLogger.Log(tag, addMoreInfo(message), throwable == null ? null : Log.getStackTraceString(throwable), logLevel, errorCode);
-                }
-            } catch (Exception e) {
-                // log message as warning to report callback error issue
-                Log.w(tag, String.format(CUSTOM_LOG_ERROR, message));
-            }
+                throw new IllegalArgumentException("Unknown logLevel");
         }
     }
 
     /**
-     * Logs debug message.
+     * @deprecated use {@link com.microsoft.identity.common.internal.logging.Logger#info(String, String, String)} instead.
```

Comment:
@addev-ashish Do we want to hedge this with some kind of messaging that external devs shouldn't use these functions at all? It's implicit in our design, but not explicitly communicated anywhere 🤔

Reply:
I'm not sure what the prevailing attitude is right now towards the removal of public API surface. If we're OK with a major-rev then I say sure, let's yank the visibility. If we need to continue to expose these methods, let's deprecate.

@addev-ashish Can you lend some guidance here? Can we be bold and start trimming surface?